Site/PullRequestPreviewBranchWorkflow.md

```diff
@@ -0,0 +1,132 @@
+# A Proposed Pull Request Workflow using a Preview Branch
+
+The primary goal of this workflow is to allow the usage of a single BSSw
+GitHub branch and Pull Request (PR) to be created and edited by multiple
+actors where all reviews, edits, and all other modifications are made in that
+one branch/PR before the PR is merged to the `master` branch and deployed on
+the `bssw.io` site and to allow for previews on `preview.bssw.io` without
+needing to merge to `master` first.  This is in contrast to current workflow
+where the draft content is merged to `master` very early and subsequent edits
+are done un-reviewed on `master`.
+
+
+## Actors in the workflow
+
+* **Author:** Creates a branch with the first draft of the proposed content on
+  a branch (i.e. `content-A` in their fork of the BSSw repo) and then creates
+  a PR of that branch against BSSw `master`.  (This could be actual content
+  author or the editorial board member that collects the initial draft of the
+  contribution from from Google Form placed there by the original author.)
+
+* **EB Member:** BSSw editorial board member that shepherds the new PR through
+  the editorial process (and may make small changes at the permission or
+  request of the original author).
+
+* **Technical Editor:** Professional technical writer that reviews and edits
+  the document.
+
+* **Sand Box:** A staff member at the company Sand Box that makes small
+  changes to the proposed content so that it is displayed properly on the
+  BSSw.io site.
+
+
+## Overview of the workflow
+
+An overview of the workflow using a `preview` branch is shown in the below
+figure.
+
+![BSSw Pull Request Preview Branch Workflow](../images/PullRequestPrerviewWorkflow.png)
+
+This is a simplification of the [throw-away integration test
+branch](https://docs.google.com/document/d/1uVQYI2cmNx09fDkHDA136yqDTqayhxqfvjFiuUue7wo#heading=h.2r0g9kvx5b2a)
+where the `preview` branch takes the place of the 'next' branch.
+
+In this setup, the `production bssw.io` site is generated from the `master`
+branch while the `preview.bssw.io` site is generated from the `preview`
+branch.
+
+Some simple rules for the workflow:
+
+* To preview the content, merge the `content-X` branch to the `preview` branch
+  and observe it on the `preview.bssw.io` site.
+
+* All edits and reviews to the proposed content are performed on the
+  `content-X` branch and the associated PR before merging to the `master`
+  branch.
+
+* When the proposed content is 100% ready to publish, the `content-X` branch
+  is "graduated" (i.e. is merged to the `master` branch by merging to
+  associated PR).
+
+* The `preview` branch always contains `master` so every merge to `master`
+  must be followed up by a merge of `master` to `preview`.
```

Comment:
The reality is that there are articles that may have been unpublished for some reason.  Right now, we do this by changing the article metadata to publish=preview or publish=no.  In a world where `master` == published and `preview` == preview instead of relying on the metadata, what do we do with unpublished articles?

Reply:
> The reality is that there are articles that may have been unpublished for some reason.  Right now, we do this by changing the article metadata to publish=preview or publish=no.  In a world where `master` == published and `preview` == preview instead of relying on the metadata, what do we do with unpublished articles?

Two options:

1) Keep using the metadata to toggle publish or unpublish on 'master'

2) Remove the *.md file (using a new commit and PR).

For the same of simplicity, I would go with option-2.  (Note, file is not really gone because it is git history **forever**.)  In any case, there is always going to need to be a new git commit to either toggle the metadata or remove the *.md file so take your pick.  This is not key to the workflow in my opinion.  Either option is okay.